Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM fedora:latest
+FROM fedora:latest as development
```

Comment:
I think it would be better to keep the environment the same between dev and production. Too much potential for weird problems by compiling stuff on Fedora and then trying to run it on centos

Reply:
I'd rather run the production container on something with LTS support. Fedora is great for development because it's up to date, but it leads to package instability. For example, the minor revision of Python changed one or two versions between the start of this repository and now. This caused issues with pipenv. We could use CentOS as the base and extend the repository sources, but Fedora has up to date tooling for things like clang, gcc, etc.

Our dependencies are well known (msgpack, nss, nspr, python36), so the wheel should be portable. I think unit tests in the development container and integration tests in the production container should help with this mismatch.

What I can do is replace Fedora with CentOS, but still keep the staged docker build.

Reply:
Yeah, replacing Fedora with CentOS was what I was thinking.

Reply:
The situation I'm trying to avoid is one where Fedora gets some distro/package update which suddenly breaks everything.

Reply:
Among such changes would be something like https://fedoraproject.org/wiki/Changes/F31_Mass_Python_2_Package_Removal. In Fedora 30, `scons` is probably an alias for `python2-scons`.

Reply:
The development image is now running centos -- the EPEL makes this a straightforward change.